internal/provider/validators.go

```diff
@@ -347,6 +349,25 @@ func validateCIDRBlock(v interface{}, path cty.Path) diag.Diagnostics {
 	return diagnostics
 }
 
+// Validate the project ID in provider and resource config
+func validateProjectID(v interface{}, path cty.Path) diag.Diagnostics {
```

Comment:
I would remove this helper and use the plugin SDK's helper instead: https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/helper/validation#IsUUID